azurerm/resource_arm_virtual_network.go

```diff
@@ -183,6 +203,8 @@ func resourceArmVirtualNetworkRead(d *schema.ResourceData, meta interface{}) err
 			d.Set("address_space", space.AddressPrefixes)
 		}
 
+		d.Set("ddos_protection_plan", flattenVirtualNetworkDDosProtectionPlan(props))
```

Comment:
As this is a complex object could we check err here?
```suggestion
		if err := d.Set("ddos_protection_plan", flattenVirtualNetworkDDosProtectionPlan(props)); err != nil {
			return fmt.Errorf("Error setting `ddos_protection_plan`: %+v", err)
		}
```